dataprofiler/profilers/helpers/report_helpers.py

```diff
@@ -100,7 +118,50 @@ def _prepare_report(report, output_format=None, omit_keys=None):
         if isinstance(value, set):
             value = sorted(list(value))
 
-        if isinstance(value, dict):
+        # For data_stats (in structured case), need to recurse through a list
+        # As well as account for omit_keys containing indices instead of keys
+        if key == "data_stats" and isinstance(value, list):
```

Comment:
So the reason for both conditions is unstructured has the key but is not a list?

Reply:
Correct